business/checkers/authorization/service_checker.go

```diff
@@ -0,0 +1,68 @@
+package authorization
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+	v1 "k8s.io/api/core/v1"
+)
+
+type ServiceChecker struct {
+	ServiceRole kubernetes.IstioObject
+	Services    []v1.Service
+}
+
+// Check verifies that the services point to existing ones. ServiceRole can only affect the defined namespace, no other even if FQDN is used to point to different namespace
+func (sc ServiceChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	if rulesSpec, found := sc.ServiceRole.GetSpec()["rules"]; found {
+		if rules, ok := rulesSpec.([]interface{}); ok {
+			for ruleIndex, ruleSpec := range rules {
+				if rule, ok := ruleSpec.(map[string]interface{}); ok {
+					if servicesSpec, found := rule["services"]; found {
+						if service, ok := servicesSpec.([]interface{}); ok {
+							for _, s := range service {
+								if host, ok := s.(string); ok {
+									fqdn := kubernetes.ParseHost(host, sc.ServiceRole.GetObjectMeta().Namespace, sc.ServiceRole.GetObjectMeta().ClusterName)
+									if fqdn.Namespace != sc.ServiceRole.GetObjectMeta().Namespace {
+										validation := models.Build("servicerole.invalid.namespace", fmt.Sprintf("spec/rules[%d]/services", ruleIndex))
+										validations = append(validations, &validation)
```

Comment:
I see you don't add the servicerole index to the path. Is there something behind that makes it better not to add it?

Reply:
Yes, the format. Multiple services are done in the yaml like this: ``services: ["service1", "service2"]``. We don't support inline paths?

Reply:
oh I see. Yep, we do cover that scenario. The yaml render it converts the inline yaml to a dashed list.

Here an example:
```yaml
apiVersion: "rbac.istio.io/v1alpha1"
kind: ServiceRole
metadata:
  name: products-viewer
  namespace: bookinfo
spec:
  rules:
  - services: ["productpage.istio-system.svc.cluster.local", "productpage", "reviews.othernamespace.svc.cluster.local"]
    methods: ["GET", "HEAD"]
```

The output of the validation is:
![Screenshot-2019-3-19 Kiali Console](https://user-images.githubusercontent.com/613814/54597245-8580a200-4a36-11e9-93ba-7acd2d3dea39.png)